### PR TITLE
fix: Better wording when sharing profile on X/Telegram

### LINKED ISSUE #10452: Better wording in "share on Twitter"
Profiles
---
Once you create your profile you can share it on Twitter:
Screenshot 2024-06-11 at 11 49 22

Text looks fugly
---
And it doesn't feel alright. 
Screenshot 2024-06-11 at 11 49 37

Solution
---
Can we have new text created, especially for profile creation sharing? 
**Something like**: I just created my profile on Koda. Create yours and follow me on.... link
feel free to suggest other wording. This is just an idea.

- check pls also if this applies on telegram share

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix

## Needs Design check

- @exezbcz please review

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #10452
- [ ] Requires deployment <worker/stick/speck>

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
- X
image

- telegram
image

### CODE REVIEW

File: components/common/successfulModal/ShareSocialsSection.vue [hunk 1]
```diff
@@ -63,7 +63,7 @@ const { toast } = useToast()
 const { shareOnX, shareOnTelegram, shareOnFarcaster } = useSocialShare()
 
 const handleShareOnX = () => {
-  shareOnX(props.text, props.url)
+  shareOnX(props.text, props.url, '')
```
Comment (hassnian) [inline components/common/successfulModal/ShareSocialsSection.vue:66]:
```suggestion
  shareOnX(props.text, props.url, null)
```

Comment (Jarsen136) [reply components/common/successfulModal/ShareSocialsSection.vue:66]: type mismatch 👀 
image

Comment (hassnian) [reply components/common/successfulModal/ShareSocialsSection.vue:66]: `via?: string = 'KodaDot'` or `via: string|null = 'KodaDot'`

it's a small thing , I would not add that query param to the url if it's nullish

File: composables/useSocialShare.ts [hunk 2]
```diff
@@ -13,12 +13,12 @@ export default function () {
...
-    via: string = 'KodaDot',
+    via: string | null = 'KodaDot',
   ) => {
     const shareUrl = new URL('https://twitter.com/intent/tweet')
-    shareUrl.searchParams.set('text', text)
-    shareUrl.searchParams.set('via', via)
-    shareUrl.searchParams.set('url', url)
+    text && shareUrl.searchParams.set('text', text)
+    via && shareUrl.searchParams.set('via', via)
+    url && shareUrl.searchParams.set('url', url)
```
Comment (hassnian) [inline composables/useSocialShare.ts:21]:
```suggestion
    shareUrl.searchParams.set('text', text)
    via && shareUrl.searchParams.set('via', via)
    shareUrl.searchParams.set('url', url)
```
was maybe mb , I meant only the `via`, even though the other could also be `''` , it's not the intent for those to be nullish, this makes more sense

commit suggestion if you agree , sorry for being picky :/

Comment (Jarsen136) [reply composables/useSocialShare.ts:21]: That's fine : )

**Review by hassnian — APPROVED** [review]

**Review by vikiival — APPROVED** [review]

**Pull request merged**